src/components/WeatherWidget.vue

```diff
@@ -1,41 +1,101 @@
 <template>
   <div class="weather-widget">
-    <WeatherWidgetPlace
-      v-bind="weatherData.location"
-      :date="selectedDayData?.date ?? 'Today'"
+    <WeatherWidgetMessage 
+      v-if="globalIsLoading"
+      :message="'Loading...'" 
     />
-    <WeatherWidgetCurrentData v-bind="selectedDayData" />
-    <WeatherWidgetForecast
-      :forecastData="weatherData.forecast"
-      :selectedDay
-      @handleDayChange="handleDayChange"
+    <WeatherWidgetMessage
+      v-else-if="globalError.length"
+      :message="globalError"
     />
+    <template v-else-if="selectedDayData && weatherData && locationKeyData">
+      <WeatherWidgetPlace
+        :placeInfo="[
+          locationKeyData.localizedName,
+          locationKeyData.country,
+          selectedDayData.date,
+        ]"
+      />
+      <WeatherWidgetCurrentData v-bind="selectedDayData" />
+      <WeatherWidgetForecast
+        :forecastData="weatherData.dailyForecasts"
+        :selectedDay
+        @handleDayChange="handleDayChange"
+      />
+    </template>
+    <p v-else>Something bad happened. Please refresh the page</p>
   </div>
 </template>
 
 <script setup lang="ts">
-import { computed, ref } from "vue";
+import { computed, ref, onMounted } from "vue";
 
-import { mockWeatherResponseData } from "@/mocks/mockWeatherDataResponse";
-import { mapWeatherData } from "@/utils/dataMappers";
-import { getCurrentDate } from "@/helpers/getCurrentDate";
+import {
+  type LocationKeyMapperOutput,
+  type WeatherDataMapperOutput,
+  mapLocationKeyData,
+  mapWeatherData,
+} from "@/utils/dataMappers";
+import { type Coordinates, getLocation } from "@/helpers/getLocation";
+import { getFormattedDate } from "@/helpers/getFormattedDate";
+import { useFetch } from "@/composables/useFetch";
+import { API_URL, endpoints } from "@/constants";
 
 import WeatherWidgetPlace from "./WeatherWidgetPlace.vue";
 import WeatherWidgetCurrentData from "./WeatherWidgetCurrentData.vue";
 import WeatherWidgetForecast from "./WeatherWidgetForecast.vue";
+import WeatherWidgetMessage from "./WeatherWidgetMessage.vue";
 
-const currentDate = getCurrentDate();
+const currentDate = getFormattedDate();
 
 const selectedDay = ref<string>(currentDate);
+const coords = ref<Coordinates | null>(null);
+const geoLocationLoading = ref<boolean>(false);
+const geoLocationError = ref<string | null>(null);
 
-const weatherData = computed(() => mapWeatherData(mockWeatherResponseData));
+const {
+  data: locationKeyData,
+  error: locationKeyError,
+  isLoading: isLoadingLocationKey,
+  fetchData: fetchLocationKey,
+} = useFetch<LocationKeyMapperOutput>(
+  `${API_URL}${endpoints.locationKey}`,
+  mapLocationKeyData
+);
+
+const {
+  data: weatherData,
+  error: weatherError,
+  isLoading: isLoadingWeatherData,
+  fetchData: fetchWeatherData,
+} = useFetch<WeatherDataMapperOutput>("", mapWeatherData);
+
+const loadingStates = computed(() => [
```

Comment:
I don't think we need 2 computed for this, we can return globalIsLoading in single one